### ISSUE TITLE
Why label_ids is part of input_ids

### ISSUE THREAD

**Issue by kgarg8** [XiaoxinHe/G-Retriever#17]
Hi,

Am I misunderstanding something or there seems a bug in the highlighted code:

https://github.com/XiaoxinHe/G-Retriever/blob/565cd9d4114e2d0817906964c15bbefc4b00f700/src/model/llm.py#L111

Why is `label_input_ids` part of `input_ids`?

**Comment by XiaoxinHe** [comment]
When fine-tuning LLaMA2 for a question-answering task, you typically concatenate the query (question) and the answer into a single sequence. The model’s input (input_ids) and labels are the concatenation of the query and the answer. The key points are:

1. **Concatenation of Query and Answer**: The input to the model is a concatenated sequence of the query followed by the answer. For example, if the query is “What is the capital of France?” and the answer is “Paris”, the concatenated input might look like this:
```
input_ids: [What, is, the, capital, of, France?, Paris]
```

2. **Labels and Loss Calculation**: The labels for the sequence are typically set such that the model is only penalized for incorrect predictions in the answer portion of the sequence. The query part of the sequence is usually ignored in the loss 
calculation. This means the loss is calculated only based on the tokens in the answer portion.

Here’s a more detailed breakdown:

- Input Sequence: [query tokens, answer tokens]
- Label Sequence: [-100, -100, ..., -100, answer tokens]

In this setup, the tokens corresponding to the query are masked out (commonly with a value like -100), so the loss is only computed for the answer tokens. This way, the model is trained to generate the correct answer given the query as context.

**Comment by kgarg8** [comment]
Thanks for the clarification.
